banyand/metadata/schema/etcd.go

```diff
@@ -192,7 +191,7 @@ func (e *etcdSchemaRegistry) get(ctx context.Context, key string, message proto.
 	return nil
 }
 
-func (e *etcdSchemaRegistry) update(ctx context.Context, metadata Metadata) error {
+func (e *etcdSchemaRegistry) update(ctx context.Context, metadata Metadata, allowOverwrite bool) error {
```

Comment:
How about splitting this into `update` and `create`. they have different preconditions.